readthedocs.yml

```diff
@@ -0,0 +1,5 @@
+conda:
+    file: docs/environment.yml
```

Comment:
It is not clear to me what this environment.yml should be? The one on which the package works or dependencies needed for the docs??

Reply:
this is an conda environment file ([example](https://github.com/psyplot/psyplot/blob/master/docs/environment.yml)) that tells readthedocs.org about the dependencies you need to build the docs (see [here](https://docs.readthedocs.io/en/stable/guides/conda.html)). You should list here additional sphinx extensions, or dependencies of your package that you do not want to have installed via `pip` (`pathos` and `pandas` for instance).

Note that, from my experience, readthedocs does not work with the conda-forge channel, as it eats too much memory. I therefore took necessary packages from the conda-forge channel and added them to my own channel which I am then using in the `environment.yml` file.

Reply:
Hmm ok thanks. Still unclear to me what you mean by your "own channel". What is this channel? Is this in github or somewhere else? Sorry if this is a stupid question :/

Reply:
but if I do not mention anything of pandas or pathos in the docs, do I still need this? I am a little confused about the purpose of this environment

Reply:
> What is this channel? Is this in github or somewhere else?

it's an anaconda channel. You can register there and publish your own packages under your own channel (here is mine: anaconda.org/chilipp/)

> but if I do not mention anything of pandas or pathos in the docs, do I still need this?

You are using the directives of `sphinx.ext.autodoc` (`.. automodule::` for instance). This will import your package when building the docs. If `pathos` is not needed for building the docs (i.e. it is not imported on a module level), you can also exclude it

Reply:
an alternative is, to import pathos only, if you are not building the docs on readthedocs. In your `core_functions.py` for instance, you could change the `pathos` import into something like

```python
import os
if not os.getenv('READTHEDOCS'):
    from pathos.multiprocessing import ProcessingPool as Pool
```

this is not a super nice solution, but it would be the easiest

Reply:
then let's do it this way for the moment and hope it works. I have now added an environment.yml file, inspired from the conda environment I used when working on the docs. I am a bit confused because autodocsumm is not there. Did I do something wrong?

Reply:
> I am a bit confused because autodocsumm is not there

not sure what you mean. Do you mean, it's not available through the `defaults` conda channel? You could list it to be installed via pip in your `environment.yml`, see https://github.com/psyplot/psyplot/blob/152bf556cc68142060ba06e91905460d74250f23/docs/environment.yml#L19 for instance

Reply:
Sorry, that was probably not clear. I meant that I created the environment directly from my working conda env (with  conda env export > environment.yml) and the resulting environment.yml does not contain autodocsumm. I found this surprising. Anyway, I'll add it then

Reply:
ah, I see :sweat_smile: then we can mark this as resolved, I suppose :smiley: